Home_Work_15/HW15.cpp

```diff
@@ -29,128 +27,123 @@ int sumStudentsPassedTreshold(Student arrayOfStudents[amountOfStudents], const i
+bool sortByAvarageRate(const Student& firstStudent, const Student& secondStudent)
 {
-    float avarageRateFirstStudent = 0;
-    float avarageRateSecondStudent = 0;
+	float avarageRateFirstStudent = 0;
+	float avarageRateSecondStudent = 0;
 
-    for (int i = 0; i < std::size(firstStudent.rates); i++)
-    {
-        avarageRateFirstStudent += firstStudent.rates[i];
-    }
+	for (int i = 0; i < std::size(firstStudent.rates); i++)
+	{
+		avarageRateFirstStudent += firstStudent.rates[i];
+	}
 
-    avarageRateFirstStudent /= static_cast<float>(std::size(firstStudent.rates));
+	avarageRateFirstStudent /= static_cast<float>(std::size(firstStudent.rates));
 
-    for (int i = 0; i < std::size(secondStudent.rates); i++)
-    {
-        avarageRateSecondStudent += secondStudent.rates[i];
-    }
+	for (int i = 0; i < std::size(secondStudent.rates); i++)
+	{
+		avarageRateSecondStudent += secondStudent.rates[i];
+	}
 
-    avarageRateSecondStudent /= static_cast<float>(std::size(secondStudent.rates));
+	avarageRateSecondStudent /= static_cast<float>(std::size(secondStudent.rates));
 
-    return avarageRateFirstStudent < avarageRateSecondStudent;
+	return avarageRateFirstStudent < avarageRateSecondStudent;
 }
 
 void printStudent(Student currenStudent)
 {
-    std::string rates = "";
-    for (int i = 0; i < std::size(currenStudent.rates); i++)
-    {
-        rates += " ";
-        rates += std::to_string(currenStudent.rates[i]);
-    }
-
-    std::cout << "Name: " << currenStudent.name << " Rates: " << rates << std::endl;
-  
+	std::string rates = "";
+	for (int i = 0; i < std::size(currenStudent.rates); i++)
+	{
+		rates += " ";
+		rates += std::to_string(currenStudent.rates[i]);
+	}
+
+	std::cout << "Name: " << currenStudent.name << " Rates: " << rates << std::endl;
+
 }
 
 Student* findBest(Student arrayOfStudents[amountOfStudents], const int amountOfStudents)
 {
-    std::vector<int> rateSums;
-    int tempSum;
-    Student* bestStudentPtr = nullptr;
-
-    for (int i = 0; i < amountOfStudents; i++)
-    {
-        tempSum = 0;
-
-        for (int j = 0; j < std::size(arrayOfStudents[i].rates); j++)
-        {
-            tempSum += arrayOfStudents[i].rates[j];
-        }
-        
-        rateSums.push_back(tempSum);
-    }
-
-    for (int i = 0; i < rateSums.size()-1; i++)
-    {
-        if (rateSums[i]> rateSums[i+1])
-        {
-            bestStudentPtr = (arrayOfStudents + i);
-            std::swap(rateSums[i], rateSums[i + 1]);
-            std::swap(arrayOfStudents[i], arrayOfStudents[i + 1]);
-        }
-        
-        else 
-        {
-            bestStudentPtr = (arrayOfStudents + (i+1));
-        }
-    }
-    return bestStudentPtr;
+	int tempSumFirstStudent = 0;
+	int tempSumSecondStudent = 0;
+	Student* bestStudentPtr = nullptr;
+
+	for (int i = 0; i < amountOfStudents - 1; i++)
+	{
+		for (int j = 0; j < std::size(arrayOfStudents[i].rates); j++)
+		{
+			tempSumFirstStudent += arrayOfStudents[i].rates[j];
+		}
```

Comment:
you can use it to compare students
`float averegeRateOfStudent(const Student& studentToClaculate);`